### ISSUE TITLE
ContentTypeRecords with duplicate names causes some problems

### ISSUE THREAD

**Issue by roballred** [OrchardCMS/Orchard#5845]
File this under: I know its stupid to do this but we should be more graceful when this happens.

Doing some testing I inadvertently added a content type into Orchard_Framework_ContentTypeRecord with the same name twice. This causes an exception in Orchard.Data.Repository starting at line 91 as follows:

        public virtual T Get(Expression<Func<T, bool>> predicate) {
            return Fetch(predicate).SingleOrDefault();
        }

The problem is that SingleOrDefault expects 0 or 1 results, which seems reasonable until you have more than one. Can we just take FirstOrDefault? At very least it seems we could have some message logged that this is happening because of duplicate content item names.

I am curious to learn what others think.

**Comment by sebastienros** [comment]
No, this is better to have an exception, otherwise you could have more issues if you don't detect it early. Like half of your content items pointing to the wrong type.

Other solution is to add a constraint to the database.
